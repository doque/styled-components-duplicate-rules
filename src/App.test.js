import React from 'react';
import styled from 'styled-components';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';

Enzyme.configure({ adapter: new Adapter() });

const InputField = styled.input`
  &:read-only,
  &:disabled {
    background: lightgray;
  }
  &::after {
    content: "no"
  }
}
`;

test('mounts correctly with readOnly prop', () => {
  const wrapper = shallow(<InputField readOnly disabled />);
  expect(wrapper).toMatchSnapshot();
  // expect(wrapper.children()).toMatchSnapshot();
});
