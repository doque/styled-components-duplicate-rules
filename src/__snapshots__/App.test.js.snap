// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mounts correctly with readOnly prop 1`] = `
.c0:-moz-read-only,
.c0:disabled {
  background: lightgray;
}

.c0:read-only,
.c0:disabled {
  background: lightgray;
}

.c0::after {
  content: "no";
}

<input
  className="c0"
  disabled={true}
  readOnly={true}
/>
`;
